### DISCUSSION TITLE
Security Group Id: sg-12345 not found

### DISCUSSION THREAD

**Discussion by Ichimokuzero** [aws/aws-cdk#21485] [Q&A]
I was trying to import an existing Security Group in CDK by using SecurityGroup.fromSecurityGroupId method and it failed with the following error:

Security Group Id: sg-12345 not found! (Service: AmazonEC2; Status Code: 400; Error Code: InvalidGroup.NotFound; Request ID: 8e2cd924-075d-4c64-b5ba-2e1d9c72fe95; Proxy: null)

Below is my CDK code:
const sg = SecurityGroup.fromSecurityGroupId(this, 'sgFromLookUp', 'sg-084c533df9d662439');

I double checked that the security group id is correct, I also tried the other 2 methods for security group look up:
SecurityGroup.fromLookupById()
SecurityGroup.fromLookupByName()

All of them returned the same error, any ideas why?

**Comment by john-aws** [comment]
I'm guessing it's related to this [dummy SG value](https://github.com/aws/aws-cdk/blob/74318c7d22bfc00de9e005f68a0a6aaa58c7db39/packages/%40aws-cdk/aws-ec2/lib/security-group.ts#L406) in CDK. You may have a dependency issue and need to make some resource dependent on the security group in question so that the security group ID can be resolved.

**Comment by jumic** [comment]
Method `SecurityGroup.fromSecurityGroupId` doesn't use Security Group `sg-12345`. This Security Group ist only part of the `fromLookup` methods.


Can you please describe your use case so we can see which of those methods is correct? Can you share the code for a minimum example where the issue occurs? (using `SecurityGroup.fromSecurityGroupId`) Please also add the error message again.

**Comment by yaseen104** [comment]
You can use `fromSecurityGroupId` . 
Working code https://kuchbhilearning.blogspot.com/2022/10/get-security-group-from-id-aws-cdk.html
